### ISSUE TITLE
Incorrect handling of escaped hash character

### ISSUE THREAD

**Issue by avikivity** [pkgconf/pkgconf#215]
If an escaped hash character appears in in a value, it is not treated as a comment introduction, but it is also removed from the output.

Example:

```
Name: x
Description: x
Version: 1.0

Cflags: kuku=\#ttt
```

```console
$ pkg-config --cflags  ./x.pc
kuku=ttt
```

This breaks Clang's `-Wno-error=#warnings` flag (it was a bad idea to use a comment character in a command line argument, ut still).

Seen with 1.7.3.

**Comment by kaniini** [comment]
If it is removed from the output, then the lexer *is* treating it as a comment introduction.

**Comment by kaniini** [comment]
anyway, is there any other implementation where that testcase passes as expected?  i am skeptical

**Comment by avikivity** [comment]
> If it is removed from the output, then the lexer _is_ treating it as a comment introduction.

Well, the remainder of the line survives. If it was a comment introduction, it's expect it to be truncated after `kuku=`.

**Comment by avikivity** [comment]
> anyway, is there any other implementation where that testcase passes as expected? i am skeptical

I wasn't able to even build pkg-config, so I can't test it. But excising random `#` characters can't be right.

**Comment by kaniini** [comment]
what a weird bug.  i'm honestly not sure what to do about this one.

**Comment by avikivity** [comment]
I tested pkg-config 0.29.1 from Ubuntu 20.04. It emits the # and even quotes it:

```console
# cat > x.pc
Name: x
Description: x
Version: 1.0

Cflags: kuku=\#ttt
# pkg-config --cflags  ./x.pc
kuku=\#ttt
```

**Comment by avikivity** [comment]
Thanks! And I'd be grateful if a patch release can be issued, so I can try to lobby my distro to update.

**Comment by kugel-** [comment]
Seems to be fixed by 52d19e1b9e7242ebba8bc54e1f69636079d98fae according to git-bisect
